app/assets/javascripts/dashboard/views/dataset/dataset_import_view.js

```diff
@@ -0,0 +1,157 @@
+// @flow
+
+import React from "react";
+import { Button, Spin, Input, Checkbox } from "antd";
+import Request from "libs/request";
+import update from "immutability-helper";
+import Toast from "libs/toast";
+import type { APIDatasetType } from "admin/api_flow_types";
+
+class DatasetImportView extends React.PureComponent {
+
+  props: {
+    datasetName: string,
+    isEditingMode: false,
+  }
+
+  state: {
+    dataset: ?APIDatasetType,
+    datasetJson: string,
+    isValidJSON: boolean,
+  } = {
+    dataset: null,
+    datasetJson: "",
+    isValidJSON: true,
+  }
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  async fetchData(): Promise<void> {
+    const datasetUrl = `/api/datasets/${this.props.datasetName}`;
+    const dataset = await Request.receiveJSON(datasetUrl);
+
+    const datasetJsonUrl = `${dataset.dataStore.url}/data/datasets/${this.props.datasetName}`;
+    const datasetJson = await Request.receiveJSON(datasetJsonUrl);
+
+    // eslint-disable-next-line react/no-did-mount-set-state
+    this.setState({
+      dataset,
+      datasetJson: JSON.stringify(datasetJson.dataSource, null, "  "),
+    });
+  }
+
+  importDataset = () => {
+    if (this.props.isEditingMode) {
+      const url = `/api/datasets/${this.props.datasetName}`;
+      Request.sendJSONReceiveJSON(url, {
```

Comment:
Correct. The result is ignored. The is the only method we have for sending JSON. The `...ReceiveJSON` ending only indicates the expected return type from the server.